encoding/thrift/inbound_test.go

```diff
@@ -130,16 +129,18 @@ func TestThriftHandler(t *testing.T) {
 		ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 		defer cancel()
 
-		handler := NewMockUnaryHandler(mockCtrl)
-		h := thriftUnaryHandler{Protocol: proto, UnaryHandler: handler, Enveloping: true}
+		handler := func(ctx context.Context, w wire.Value) (Response,
+			error) {
 
-		if tt.expectHandle {
-			handler.EXPECT().Handle(gomock.Any(), requestBody).
-				Return(Response{
-					Body:               fakeEnveloper(tt.responseEnvelopeType),
-					IsApplicationError: tt.responseIsAppError,
-				}, nil)
+			if tt.expectHandle {
+				assert.Equal(t, requestBody, w)
```

Comment:
Not what this was testing. This was testing that if `tt.expectHandle` is true,
then this function is called with these arguments. Otherwise, it fails if this
function is called.

You should just make the function check the requestBody, etc. and track
whether the function was called in a boolean outside. Then just match that
bool against `tt.expectHandle`.